### DISCUSSION TITLE
16x RC Servo Phidget Connection & Control

### DISCUSSION THREAD

**Discussion by Helmi8800** [artisan-roaster-scope/artisan#1771] [Q&A]
Hello guys. My name is Helmi. .I'm working on to control servo motor using 16x RC Servo Phidget wtih VINT Hub0001_0. I tried everything that been suggested in artisan scope blog but could not find any success in control the servor. But i can still control servor motor using Phidget control panel. Anyone can assist me to figure out right setup and rc command code to control the servo motor?. Thanks!

Here screen shot of my setting 
<img width="658" alt="Screenshot 2024-12-29 at 5 02 43 PM" src="https://github.com/user-attachments/assets/24b94ede-5f4c-467a-9289-db91d6c857ab" />
<img width="923" alt="Screenshot 2024-12-29 at 5 03 40 PM" src="https://github.com/user-attachments/assets/6d026d4e-e361-4add-9adc-02de521281aa" />
<img width="913" alt="Screenshot 2024-12-29 at 5 03 54 PM" src="https://github.com/user-attachments/assets/4ac6bf98-b353-4dbf-82bd-5a1a5ce43e2d" />
<img width="911" alt="Screenshot 2024-12-29 at 5 04 09 PM" src="https://github.com/user-attachments/assets/b744b6db-e573-44e3-a237-b3c17daec665" />
<img width="917" alt="Screenshot 2024-12-29 at 5 04 18 PM" src="https://github.com/user-attachments/assets/36148dd6-59a0-4fc9-94be-35878a9cc57b" />
<img width="1421" alt="Screenshot 2024-12-29 at 5 05 17 PM" src="https://github.com/user-attachments/assets/9b4efc3a-d17c-433c-9412-9b2bfbe53118" />
<img width="1421" alt="Screenshot 2024-12-29 at 5 06 25 PM" src="https://github.com/user-attachments/assets/b8f2ebf1-70f1-430d-9cd9-697dc72df9a0" />
![IMG_1217](https://github.com/user-attachments/assets/b778eeb5-4956-4554-bbc6-ce2ebe41caf0)

**Comment by roasterdave** [comment]
Is the Phidget control panel closed when you run Artisan?

**Reply by Helmi8800** [reply]
yes, already closed. is there any port configuration needed to run it with macbook?

**Reply by MAKOMO** [reply]
Can you control the servo using the Phidget Control Panel? 

Your configuration looks ok. Any hint in `~/Library/Application Support/artisan-scope/Artisan/artisan.log` file?

**Reply by hermetheuscoffee** [reply]
Hi, @Helmi8800 I know this is an older thread, but I've used the exact setup you have pictured. Your Artisan config looks right, but your servo is plugged in backwards. :-) Your orange/yellow wire (which is the pulse width wire) should be closest to the edge of the board. Of course, I don't know how you would have managed to get the Phidget control panel to control it this way, but it's an issue nonetheless. 
Beyond that, make sure that the Phidget control panel is FULLY closed (it can run in the background without you knowing). Use Task Manager in Windows to ensure it's not running. Only one program can have connection to the Phidget, so if that's running in the background, Artisan can't take control.

**Reply by Helmi8800** [reply]
Hello guys, after many years. I finally solved the problem. Since the connection using Pdhiget hub, i need to insert Hub Serial Number. Then artisan can find  the hub > channel output > rc motor channel . Here my command for slider 

Slider
set(0,{},671950);engaged(0,1,671950);

Button
set(0,{},671950);engaged(0,1,671950);

Hope this will help all artisan user who want to automate their machine using phidget :)
